### PR TITLE
[fix](memory leak) Fix load fragment `QueryFragmentsCtx` is not destroyed

### PR DESCRIPTION
**Pull request by xinyiZzz**
# Proposed changes

Issue Number: close #xxx

## Problem summary

For load fragments, the fragment_num sent by FE has a small probability of 0.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.95 seconds
 load time: 477 seconds
 storage size: 17123356274 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221205202544_clickbench_pr_58243.html

File: be/src/runtime/fragment_mgr.cpp [hunk 1]
```diff
@@ -510,7 +510,10 @@ void FragmentMgr::_exec_actual(std::shared_ptr<FragmentExecState> exec_state, Fi
         std::lock_guard<std::mutex> lock(_lock);
         _fragment_map.erase(exec_state->fragment_instance_id());
         if (all_done && fragments_ctx) {
-            _fragments_ctx_map.erase(fragments_ctx->query_id);
+            auto search = _fragments_ctx_map.find(fragments_ctx->query_id);
```
Comment (yiguolei) [inline be/src/runtime/fragment_mgr.cpp:513]: why search it first? I think map.erase method will do this automanticly.

Comment (xinyiZzz) [reply be/src/runtime/fragment_mgr.cpp]: Indeed, fix

**Review by yiguolei — DISMISSED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**